### DISCUSSION TITLE
[Feature]: Nachladen von Nachrichten

### DISCUSSION THREAD

**Discussion by techotline** [lotta-schule/web#240] [Ideen]
### Bereich

Nachrichten (inkl. "Meine Beiträge")

### Verhalten

Verhalten

### Problem

Idee: Da die Flut an täglichen Nachrichten massiv zunimmt, wäre eine "Zeitbegrenzung", welche die maximale Dauer der Speicherung von NACHRICHTEN einschießt, eine effiziente Variante bzw. Methode, um innerhalb einer Nachrichtenperiode (z.B. Schuljahr X) Gruppen-Chatverläufe übersichtlicher zu gestalten und "Unwichtiges" zeitnah zu löschen.

### Beschreibung

Lösung beinhaltet zwei neue Funktionen im Dashboard:

- der Administrator kann selbstlöschende Nachrichten für eine Gruppe konfigurieren
- der Administrator kann die Dauer festlegen, in dem er die gewünschten Attribute (Dauer bevor Nachrichten "verschwinden")  über ein Input-Feld selbst festlegt

-> Nachrichten werden nach _ _100 Tagen_ _ gelöscht.

**Comment by ptitmouton** [comment]
Stimme zu dass wir Lösungen finden müssen, um den Nutzern mehr Kontrolle über den eigenen Nachrichtenchat zu geben.
Halte selbstlöschende Nachrichten für keine gute Lösung. 
Es ist nicht mehr, sondern weniger Kontrolle. Ich denke wir sollten eher in die Richtung gehen, Konversationen stummzuschalten und zu "verstecken" (in ein Untermenü zu verbannen oder eine Person aka Konversation quasi zu "blocken").

Mir ist auch nicht ganz klar, wie "selbstlöschende Nachrichten" gegen "die Flut an täglichen Nachrichten" hilft.

**Reply by techotline** [reply]
Gutes Argument. Ich sehe ein, dass dieser Prozess sich mit #241 in die Quere kommt. Sozusagen überflüssig. Diese Idee sollte vom Tisch sein. 

Zu Deiner Frage: Gruppen werden nur einmal im Jahr entfernt, womit auch alle verfassten Nachrichten zu diesem Zeitpunkt verschwinden. Hat der Chat ein großes Volumen an Nachrichten, wird er auch automatisch länger geladen. Warum nicht kompensieren (bzw. einstellen können), dass veraltete oder gelöste Anliegen bereits vorher für alle "verloren" gehen, um die Unterhaltung übersichtlicher zu gestalten?

Andere Meinungen sind willkommen.

**Reply by ptitmouton** [reply]
Gruppen werden nicht ‚einmal im Jahr gelöscht‘. Nur weil manche Schulen das tun und so den Verlust von Nachrichten aktiv in Kauf nehmen ist das keine Regel. Das ist dann die Entscheidung derer, die das entsprechende Lotta administrieren. Und auch dann geht es nur um Gruppennachrichten. Was ist mit 1:1-Nachrichten?

Falls es hier die ganze Zeit nur darum geht, dass die Konversations-Seite langsamer wird wenn viele Nachrichten vorhanden sind, kommt mir das eher so vor als sollten wir das mal angehen, anstatt hier künstlich komplizierte neue Features draufzuschmeißen.

**Reply by BillyTheKing** [reply]
Vorschlag: ein Vor-Laden der letzten 50 Nachrichten und wenn man noch ältere Nachrichten sehen möchte, entsprechend zu swipen oder zu klicken.

**Comment by ptitmouton** [comment]
Lösung:
Nachladen von Nachrichten

**Comment by ptitmouton** [comment]
Es gibt jetzt hierzu ein Ticket: #291 

Ich denke diese Diskussion kann damit vorerst geschlossen werden.
